crates/topos-tce-broadcast/src/double_echo/broadcast_state.rs

```diff
@@ -57,28 +63,36 @@ impl BroadcastState {
             });
         }
 
-        state.update_status();
+        state.update_status().await;
 
         state
     }
 
-    pub fn apply_echo(&mut self, peer_id: PeerId) -> Option<Status> {
+    pub async fn apply_echo(&mut self, peer_id: PeerId) -> Option<Status> {
         self.subscriptions_view.echo.remove(&peer_id);
-        self.update_status()
+        self.update_status().await
     }
 
-    pub fn apply_ready(&mut self, peer_id: PeerId) -> Option<Status> {
+    pub async fn apply_ready(&mut self, peer_id: PeerId) -> Option<Status> {
         self.subscriptions_view.ready.remove(&peer_id);
-        self.update_status()
+        self.update_status().await
     }
 
-    fn update_status(&mut self) -> Option<Status> {
+    async fn update_status(&mut self) -> Option<Status> {
         // Nothing happened yet, we're in the initial state and didn't Procced
         // any Echo or Ready messages
         // Sending our Echo message
         if let Status::Pending = self.status {
             _ = self.event_sender.try_send(ProtocolEvents::Echo {
                 certificate_id: self.certificate.id,
+                signature: sign_message(
+                    self.validator_id.clone(),
+                    self.certificate.id,
+                    self.wallet.clone(),
+                )
```

Comment:
I think `update_status` is a fairly hot function yes? Perhaps it makes sense to look into avoiding these clones and optimize `sign_message` a bit. Perhaps we can sign with `self.wallet` directly?